src/lib/peers/turn/TurnClient.ts

```diff
@@ -0,0 +1,313 @@
+// adapted from https://github.com/advance512/nat-traversal
```

Comment:
Move these comments into the relevant issues instead.

And I'd like to see discussions on the tradeoffs and designs these people made with respect to the TURN and NAT standards.